src-electron/db/query-attribute.js

```diff
@@ -247,7 +283,7 @@ async function selectAttributeDetailsFromAllEndpointTypesAndClustersUtil(
   GROUP BY ATTRIBUTE.NAME
         `
     )
-    .then((rows) => rows.map(attributeExportMapping))
+    .then((rows) => rows.map(dbMapping.map.endpointTypeAttributeExtended))
```

Comment:
But this does not use the same thing for TYPE that the other consumer of endpointTypeAttributeExtended does.... Why not?

Reply:
How are consumers supposed to work with "type" if it means different things for the same objects depending on how you got them?  Consumers have no idea what these things look like under the hood, and its easy to end up with bad mistakes.

Maybe the answer needs to be that we deprecate "type" on attributes and define new field names that are sane and consistent?  Not necessary to block this PR on that, I guess, but we should document the discrepancies in the code and get issues filed.  But that's likely to make it impossible to work in a consistent way with attributes and things like command fields too, though I think they may already mean different things by "type" right now....

This is not a theoretical problem: the "type" weirdness on attributes led to a _lot_ of bugs in Matter until we caught (we think) all the issues.

Reply:
And just to be clear: from my point of view, Zigbee Pro constraints are not a reason for _Matter_ to have problems.  We need to make sure that all Matter helpers are consistent, and the ones that do different things that Zigbee Pro needs should not be exposed to Matter at all.

Reply:
I forgot what we were doing in attribute map. the endpoint type extended map now does the same thing as attribute map. Type will be based off of array_type and type now. See my change in db-mapping.